src/averageAges.js

```diff
@@ -55,7 +68,20 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    onlyWithSon
+      ? child.sex === 'm' && people.some(mother =>
+        mother.name === child.mother)
+      : people.some(mother => mother.name === child.mother)
+  );
+
+  const averageAgeDiff = children.reduce((aveAge, child) => {
+    const mum = people.find(mother => mother.name === child.mother);
+
+    return aveAge + ((child.born - mum.born));
+  }, 0) / children.length;
```

Comment:
your algorithm is hard, bc you have loop in loop
split this reduce to two separate operation

Reply:
Can you help me with this? There is no one mentors in the chat

Reply:
![Uploading Screenshot 2023-03-01 at 15.12.24.png…]()